src/test/java/com/pylon/spokestack/google/GoogleSpeechRecognizerTest.java

```diff
@@ -0,0 +1,152 @@
+package com.pylon.spokestack.google;
+
+import java.util.*;
+import java.nio.ByteBuffer;
+
+import org.junit.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.mockito.ArgumentCaptor;
+import static org.mockito.Mockito.*;
+
+import com.google.api.gax.rpc.ApiStreamObserver;
+import com.google.api.gax.rpc.BidiStreamingCallable;
+import com.google.cloud.speech.v1.SpeechClient;
+import com.google.cloud.speech.v1.SpeechSettings;
+import com.google.cloud.speech.v1.stub.SpeechStub;
+import com.google.cloud.speech.v1.StreamingRecognizeRequest;
+import com.google.cloud.speech.v1.StreamingRecognizeResponse;
+import com.google.cloud.speech.v1.StreamingRecognitionResult;
+import com.google.cloud.speech.v1.SpeechRecognitionAlternative;
+
+import com.pylon.spokestack.OnSpeechEventListener;
+import com.pylon.spokestack.SpeechConfig;
+import com.pylon.spokestack.SpeechContext;
+
+public class GoogleSpeechRecognizerTest implements OnSpeechEventListener {
+    private SpeechContext.Event event;
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testRecognize() throws Exception {
+        SpeechContext context = createContext();
+        MockSpeechClient client = spy(MockSpeechClient.class);
+        GoogleSpeechRecognizer recognizer =
+            new GoogleSpeechRecognizer(createConfig(), client);
+
+        // inactive
+        recognizer.process(context, context.getBuffer().getLast());
+        verify(client.getRequests(), never())
+            .onNext(any(StreamingRecognizeRequest.class));
+
+        // active/buffered frames
+        context.setActive(true);
+        recognizer.process(context, context.getBuffer().getLast());
+        verify(client.getRequests(), times(context.getBuffer().size() + 1))
+            .onNext(any(StreamingRecognizeRequest.class));
+
+        // subsequent frame
+        reset(client.getRequests());
+        recognizer.process(context, context.getBuffer().getLast());
+        verify(client.getRequests())
+            .onNext(any(StreamingRecognizeRequest.class));
+
+        // complete
+        context.setActive(false);
+        recognizer.process(context, context.getBuffer().getLast());
+        verify(client.getRequests())
+            .onCompleted();
+
+        // responses
+        client.getResponses().onNext(StreamingRecognizeResponse.newBuilder()
+            .addResults(StreamingRecognitionResult.newBuilder()
+                .addAlternatives(SpeechRecognitionAlternative.newBuilder()
+                    .setTranscript("test")
+                    .setConfidence((float)0.75)
+                    .build())
+                .build())
+            .build()
```

Comment:
https://www.forbes.com/sites/outofasia/2018/02/28/dutertes-ambitious-build-build-build-project-to-transform-the-philippines-could-become-his-legacy/#6ea70a071a7f